### PR TITLE
[platform] add Telink eagle platform support

### PR DESCRIPTION
**Pull request by itianyy**
porting openthread to Telink eagle platform

### CODE REVIEW

File: examples/Makefile-eagle [hunk 1]
```diff
@@ -0,0 +1,258 @@
+#
```
Comment (wgtdkp) [inline examples/Makefile-eagle:1]: Not necessary to be addressed in this PR, please consider supporting `CMake` too.

File: examples/Makefile-eagle [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+#  Copyright (c) 2017, The OpenThread Authors.
```
Comment (wgtdkp) [inline examples/Makefile-eagle:2]:
```suggestion
#  Copyright (c) 2020, The OpenThread Authors.
```

File: examples/platforms/eagle/alarm.c [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+#include "platform-eagle.h"
+
+
+static volatile uint32_t sTime      = 0;
+static uint32_t          sAlarmTime = 0;
+static uint32_t last_tick = 0;
+
+//tianyy
+#if 0
```
Comment (wgtdkp) [inline examples/platforms/eagle/alarm.c:50]: Remove dead code?

File: examples/platforms/eagle/alarm.c [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    return t_ms + tick/16000;
+}
+
+void EagleAlarmProcess(otInstance *aInstance)
+{
+    //begin
+    uint32_t t = sys_get_stimer_tick();
+    if(t < last_tick)
+    {
+        sTime += (0xffffffff/16000);//(0xffffffff)/16/1000
```
Comment (wgtdkp) [inline examples/platforms/eagle/alarm.c:71]: ditto.

File: examples/platforms/eagle/alarm.c [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+#endif
+
+static inline uint32_t GetCurrentMs(uint32_t t_ms,uint32_t tick)
+{
+    return t_ms + tick/16000;
+}
+
+void EagleAlarmProcess(otInstance *aInstance)
+{
+    //begin
```
Comment (wgtdkp) [inline examples/platforms/eagle/alarm.c:67]:
```suggestion
    // Begin
```

File: examples/platforms/eagle/alarm.c [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+void otPlatAlarmMilliStop(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    sAlarmTime = 0;
+}
+
+#if 0
```
Comment (wgtdkp) [inline examples/platforms/eagle/alarm.c:114]: ditto.

File: examples/platforms/eagle/alarm.c [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    //config a timer(1 ms)
+    timer_set_mode(TIMER0,TIMER_MODE_SYSCLK,0,sys_clk.pclk*1000);
+	timer_start(TIMER0);
+    plic_interrupt_enable(IRQ4_TIMER0);
+}
+#endif
+
+static inline uint32_t GetCurrentMs(uint32_t t_ms,uint32_t tick)
+{
+    return t_ms + tick/16000;
```
Comment (wgtdkp) [inline examples/platforms/eagle/alarm.c:62]:
```suggestion
    return t_ms + tick / 16000;
```

You can run `./script/make-pretty clang` to format your code.

File: examples/platforms/eagle/common/common.h [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+ * common.h
+ *
+ *  Created on: May 23, 2020
+ *      Author: Telink
+ */
+
+#ifndef COMMON_H_
+#define COMMON_H_
+#include "assert.h"
+#include "bit.h"
```
Comment (wgtdkp) [inline examples/platforms/eagle/common/common.h:11]:
```suggestion

#include "bit.h"
```

**Comment by wgtdkp** [comment]
Please help remove the `libopenthread-eagle.a` file.

**Comment by itianyy** [comment]
close this PR and reopen it using another account (yiyang.tian@telink-semi.com) because of the CLA.

**Pull request closed without merging**